### ISSUE TITLE
how to include an abstract...

### ISSUE THREAD

**Issue by olearydj** [nmfs-opensci/quarto_titlepages_v1#3]
## Question
Last one, I promise.

I'm struggling to get the `abstract` environment working with PDF book formats.

It works fine for the base Quarto PDF, but when I try to use it with a `book` or `scrbook` document class, as follows, or with your templates, I get:  `LaTeX Error: Environment abstract undefined.` wherever `\begin{abstract}` is encountered.

```yaml
format:
  pdf:
    documentclass: book  % or scrbook
abstract: This is my abstract.
```

I've tried this a number of ways as I am working with partials, etc., but can't get it to work in any configuration. Here are the minimal steps to reproduce the problem:
1. Create a new Quarto document, choose type PDF
2. Add `abstract: This is my abstract.` to the YAML frontmatter.
3. Render. The PDF is output without error and it includes the abstract.
4. Change the format YAML to match the prior code block, adding `documentclass: book` or `scrbook`.
5. Render. Note that the error occurs as described on `\begin{abstract}`.

Any help would be appreciated. Thanks for your consideration.

PS - This question is also posted to the RStudio/Posit community page:
https://community.rstudio.com/t/quarto-pdf-book-scrbook-environment-abstract-undefined/154982

**Comment by eeholmes** [comment]
Ok, so what happening is that pandoc would normally include the abstract in the before-body.tex template partial, but that template partial is being swapped out by the 
```
    template-partials:
      - "before-body.tex"
      - "_titlepage.tex" 
```
files which don't have abstract included. There are many ways that you could add the abstract. Check out the `before-body.tex` that shows one way you might include it in the frontmatter. Here I added abstract to the yaml and to get that into `before-body.tex`, I use `$abstract$`. Alternatively, you could hard code (meaning paste in) the abstract text into `before-body.tex` and not include in your yaml.

**Comment by olearydj** [comment]
Thank you for your fixes and feedback!

I had figured out how partials work, but was running into trouble with environment conflicts. Hard-coding it into `before-body.tex` worked with your `_titlepage.tex`, but with mine would throw `LaTeX Error: \begin{titlepage} on input line 230 ended by \end{frontmatter}.`

I tried a few things but ultimately moved `\begin{titlepage}` and `\end{titlepage}` out of `before-body` and into my title page. That seems to have fixed it - my title page and abstract are both working now.

Again, thank you for the help!
